keyboards/keychron/q1/rev_0102/keymaps/via/keymap.c

```diff
@@ -83,6 +83,24 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
                 host_consumer_send(0);
             }
             return false;  // Skip all further processing of this key
+        case KC_TASK:
+            if (record->event.pressed) {
+                register_code(KC_LWIN);
+                register_code(KC_TAB);
+            } else {
+                unregister_code(KC_LWIN);
+                unregister_code(KC_TAB);
+            }
+            return false;  // Skip all further processing of this key
+        case KC_FLXP:
+            if (record->event.pressed) {
+                register_code(KC_LWIN);
+                register_code(KC_E);
+            } else {
+                unregister_code(KC_LWIN);
+                unregister_code(KC_E);
+            }
+            return false;  // Skip all further processing of this key
```

Comment:
These can be replaced with `G(KC_TAB)` and `G(KC_E)` directly in the keymap - or you can alias them with `#define KC_TASK G(KC_TAB)` if you want.

Reply:
Hi, We do this just to display 'TASK' and 'FILE' in VIA instead of the long key combination name, Like that:
![image](https://user-images.githubusercontent.com/66767061/143797477-b4b16a9a-3779-42f4-8c5f-6c6be6e3e36d.png)

Reply:
To be blunt, that's not really a good reason to use a custom macro when the functionality is supported out of the box.

Reply:
> 
The shortcoming of `G(KC_TAB)` and `G(KC_E)` is user can't restore these function if it have been assign to other keycode，or reassign these function to other keys by Via.

Reply:
Not intuitively, no.  But you can use the "any" key in VIA, and type in those values.